json/pickler/src/main/scala/sttp/tapir/json/pickler/Pickler.scala

```diff
@@ -104,19 +109,31 @@ object Pickler:
 
   inline given nonMirrorPickler[T](using PicklerConfiguration, NotGiven[Mirror.Of[T]]): Pickler[T] =
     summonFrom {
-      // It turns out that summoning a Pickler can sometimes fall into this branch, even if we explicitly state that we wan't a NotGiven in the method signature
-      case m: Mirror.Of[T] =>
-        errorForType[T]("Failed to summon a Pickler[%s]. Try using Pickler.derived or importing sttp.tapir.json.pickler.generic.auto.*")
       case n: NotGiven[Mirror.Of[T]] =>
         Pickler(
           new TapirPickle[T] {
-            // Relying on given writers and readers provided by uPickle Writers and Readers base traits
-            // They should take care of deriving for Int, String, Boolean, Option, List, Map, Array, etc.
-            override lazy val reader = summonInline[Reader[T]]
-            override lazy val writer = summonInline[Writer[T]]
+            override lazy val reader = summonFrom {
+              case r: Reader[T] => r
+              case _ =>
+                errorForType[T](
+                  "Use Pickler.derive[%s] instead of nonMirrorPickler. This method has to be in scope to resolve predefined picklers."
```

Comment:
is `nonMirrorPickler` something people should ever use direclty, or our impl detail?

Reply:
Yes, but it has to be in public scope. Without it Pickler has issues with deriving `ReadWriter` for `String` and some other types. This probably may be worked around, but for now I'd keep it public.